nengo/neurons.py

```diff
@@ -258,6 +258,40 @@ def step_math(self, dt, J, spiked, voltage, refractory_time):
         refractory_time[spiked > 0] = self.tau_ref + spiketime
 
 
+class PerfectLIF(LIF):
+    """Spiking version of the leaky integrate-and-fire (LIF) neuron model."""
+
+    def step_math(self, dt, J, spiked, voltage, refractory_time):
+        # reduce all refractory times by dt
+        refractory_time -= dt
+
+        # compute effective dt for each neuron, based on remaining time.
+        # note that refractory times that have completed midway into this
+        # timestep will be given a partial timestep, and moreover these will
+        # be subtracted to zero at the next timestep (or reset by a spike)
+        delta_t = (dt - refractory_time).clip(0, dt)
+
+        # update voltage using discretized lowpass filter
+        # since v(t) = v(0) + (J - v(0))*(1 - exp(-t/tau)) assuming
+        # J is constant over the interval [t, t + dt)
+        voltage -= (J - voltage) * np.expm1(-delta_t / self.tau_rc)
+
+        # determine which neurons spiked (set them to 1/dt, else 0)
+        spiked_mask = voltage >= 1
```

Comment:
Will the lowpass filter round up eventually? I just played around with it a bit, and I think it depends on `dt` and `tau_rc`. Specifically, if `np.abs(np.expm1(-dt / tau_rc)) > 0.5`, then it will round up (which is the case for `dt / tau_rc > 0.66` or so), but otherwise it will never round up.

Anyway, changing it is fine. Theoretically, it doesn't make any difference, and numerically it should hardly make a difference either.